controllers/dpa_controller.go

```diff
@@ -87,6 +88,9 @@ func (r *DPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		return result, nil
 	}
 
+	// set client to pkg/client for use in non-reconcile functions
+	oadpClient.SetClient(r.Client)
```

Comment:
Do we need this ?

Reply:
When I moved functions out of `controllers/` into `/pkg/` they lose access to clients.

I think this is a pretty clean way to make sure every `/pkg` can reuse clients.